tests/unit/iontests.js

```diff
@@ -11,126 +11,376 @@
+            'good/equivs/strings.ion', //triplequote interaction with span and whitespace corrupts the state of the parser.
+            'good/timestamp/timestamps.ion', //timestamp is not spec compliant.
+
+        ];
+
+        // For debugging, put single files in this list to have the test run only
+        // that file.   (But don't forget to clean this up on checkin!)
+        var debugList = [
+            //'good/sexps.ion'
+        ];
+
+        var goodUnskipped = [];
+        for (var path of goodAccumulator) {
+            var spath = path.replace(/\\/g, "/");
+            var shouldSkip = false;
+            for (var skip of skipList) {
+                if (spath.endsWith(skip)) {
+                    shouldSkip = true;
+                    break;
+                }
+            }
+            if (debugList.length > 0) {
+                shouldSkip = true;
+                for (var debug of debugList) {
+                    if (spath.endsWith(debug)) {
+                        shouldSkip = false;
+                        break;
+                    }
+                }
+            }
+            if (!shouldSkip) {
+                goodUnskipped.push(path);
+            }
+        }
+
+        var badUnskipped = [];
+        for (var path of badAccumulator) {
+            var spath = path.replace(/\\/g, "/");
+            var shouldSkip = false;
+            for (var skip of skipList) {
+                if (spath.endsWith(skip)) {
+                    shouldSkip = true;
+                    break;
+                }
+            }
+            if (debugList.length > 0) {
+                shouldSkip = true;
+                for (var debug of debugList) {
+                    if (spath.endsWith(debug)) {
+                        shouldSkip = false;
+                        break;
+                    }
+                }
+            }
+            if (!shouldSkip) {
+                badUnskipped.push(path);
+            }
         }
-      }
-      if (!shouldSkip) {
-        unskipped.push(path);
-      }
-    }
 
-    var suite = {
-      name: 'Good tests',
-    };
-
-    function exhaust(reader) {
-      var tries = 0;
-      for (;;) {
-        // Safety valve
-        tries++;
-        if (tries > 1000) {
-          throw new Error("Reader spinning forever!");
+        var goodSuite = {
+            name: 'Good tests'
+        };
+        var badSuite = {
+            name: 'Bad tests'
+        };
+
+        var eventStreamSuite = {
+            name: 'EventStream tests'
+        };
+        var badEventStreamSuite = {
+            name: 'Bad EventStream tests'
+        };
+
+        function goodExhaust(reader) {
+            var tries = 0;
+            for (;;) {
+                // Safety valve
+                tries++;
+                if (tries > 1000) {
+                    throw new Error("Reader spinning forever!");
+                }
+
+                var next = reader.next();
```

Comment:
Should this be a `let`? `var` is going to hoist this to the top of the function.